test/release/examples/primers/ranges.chpl

```diff
@@ -7,245 +7,384 @@
+// example, the range ``3..<10`` is equivalent to ``3..9``.  This form
+// is particularly useful for 0-based indexing, to avoid bounds like
+// ``n-1``.
+writeln("Open interval ranges");
+const rOpen = 1..<10,  // 1, 2, 3, ..., 9
+      rOpen2 = 0..<n;  // 0, 1, 2, ..., n-1
 writeRange(rOpen);
+writeRange(rOpen2);
+writeln();
+
+
+// Note that whenever a range's low bound exceeds its high bound, the
+// range is considered to be empty.
+writeln("Empty ranges");
+const empty1 = 1..0,
+      empty2 = 0..-1,
+      empty3 = 10..1;
+writeRange(empty1);
+writeRange(empty2);
+writeRange(empty3);
+writeln();
+
+// To create a range representing a decreasing series of values, the
+// ``by`` operator can be used with a negative stride.  As a simple
+// example, the following range represents the integers counting from
+// 10 down to 1.
+
+writeln("Decreasing range");
+const countDown = 1..10 by -1;  // 10, 9, 8, ..., 1
+writeRange(countDown);
+writeln();
 
-// This form can be convenient when iterating over a 0-based
-// data structure using ranges like ``lo..<data.size``.
+// See the `Range Operators` section below for more information about
+// strided ranges and the ``by`` operator.
+
+/*
+   Uses of Ranges
+   --------------
+*/
 
 //
-// Ranges are a basic building block for iteration.  Here we use a
-// for-loop to iterate over the sequence represented by ``r`` and
-// compute the sum of the indices in the sequence.
+// Ranges are a basic building block for iteration.  The following
+// for-loop iterates over the sequence represented by ``r``, computing
+// the sum of the values in the sequence.
 //
 var sum: int = 0;
-for i in r do
+for i in 1..10 do       // compute 1 + 2 + 3 + ... + 10
   sum += i;
-writeln("The sum of the values in ", r, " is ", sum);
-
-//
-// We can compute this same sum with a reduction over the range.
-//
-sum = + reduce r;
-writeln("The sum of the values in ", r, " as computed by reduce is also ", sum);
+writeln("The sum of the values in '1..10' is ", sum);
 writeln();
 
+// Ranges can also drive parallel loop forms such as
+// :ref:`forall<primers-forallLoops>` and
+// :ref:`coforall<primers-taskparallel-coforall>` loops.
+
 //
-// Ranges are also the building blocks for domains.  Here
-// we use the range ``r`` to build up a 2-D domain and an array.
+// Ranges also serve as the building block for rectangular
+// :ref:`domains <primers-domains>` and :ref:`arrays
+// <primers-arrays>`.  Here we use the range ``r`` to declare a 2D
+// 10x10 domain and an array over it.
 //
 writeln("Domains and arrays");
-const D: domain(2) = {r, r};
-var A: [D] real;
+const D = {1..10, 1..10};  // could also use '= {r, r};'
+var A: [D] real;           // a 10x10 array of real floating point values
 writeln("D = ", D);
 writeln("Array A");
 writeln(A);
 writeln();
 
+// Arrays can also be declared in terms of anonymous domains by using
+// ranges directly (optionally dropping the curly brackets):
```

Comment:
This sentence sounds like it is highlighting the concept of an "anonymous domain", which  it obviously shouldn't. Just dropping "in terms of anonymous domains" is an improvement to me. We can work "anonymous domain" in in some other fashion if you prefer, but I don't.

Reply:
That's a good point, though I do like making it clear that there's still a domain here, just one that's not named/doesn't look like one.  What about something like:

```
// Arrays can also be declared using ranges directly, which                     
// effectively creates an anonymous domain for the array:                                     

var A1: [1..n] int,
    A2: [1..n, 1..n] string;
```